core/src/main/scala/scalaz/Alt.scala

```diff
@@ -0,0 +1,111 @@
+ */
+////
+trait Alt[F[_]] extends Applicative[F] with Derives[F] { self =>
+  ////
+
+  def alt[A](a1: =>F[A], a2: =>F[A]): F[A]
+
+  // FIXME: add some, many, optional
+
+  def altly1[Z, A1](a1: =>F[A1])(f: A1 => Z): F[Z] = map(a1)(f)
+  def altly2[Z, A1, A2](a1: =>F[A1], a2: =>F[A2])(f: A1 \/ A2 => Z): F[Z] =
+    map(alt(map(a1)(-\/[A1, A2](_)), map(a2)(\/-[A1, A2](_))))(f)
+
+  def altly3[Z, A1, A2, A3](a1: =>F[A1], a2: =>F[A2], a3: =>F[A3])(
+    f: A1 \/ (A2 \/ A3) => Z
+  ): F[Z] = altly2(a1, either2(a2, a3))(f)
+  def altly4[Z, A1, A2, A3, A4](a1: =>F[A1], a2: =>F[A2], a3: =>F[A3],a4: =>F[A4])(
+    f: A1 \/ (A2 \/ (A3 \/ A4)) => Z
+  ): F[Z] = altly2(a1, either2(a2, either2(a3, a4)))(f)
+  // ... altlyN
+
+  // equivalent of tupleN
+  def either2[A1, A2](a1: =>F[A1], a2: =>F[A2]): F[A1 \/ A2] =
+    altly2(a1, a2)(identity)
+  // ... eitherN
+
+  final def altlying1[Z, A1](
+    f: A1 => Z
+  )(implicit a1: F[A1]): F[Z] =
+    altly1(a1)(f)
+  final def altlying2[Z, A1, A2](
+    f: A1 \/ A2 => Z
+  )(implicit a1: F[A1], a2: F[A2]): F[Z] =
+    altly2(a1, a2)(f)
+  final def altlying3[Z, A1, A2, A3](
+    f: A1 \/ (A2 \/ A3) => Z
+  )(implicit a1: F[A1], a2: F[A2], a3: F[A3]): F[Z] =
+    altly3(a1, a2, a3)(f)
+  final def altlying4[Z, A1, A2, A3, A4](
+    f: A1 \/ (A2 \/ (A3 \/ A4)) => Z
+  )(implicit a1: F[A1], a2: F[A2], a3: F[A3], a4: F[A4]): F[Z] =
+    altly4(a1, a2, a3, a4)(f)
+  // ... altlyingX
+
+  override def xproduct0[Z](z: =>Z): F[Z] = pure(z)
+  override def xproduct1[Z, A1](a1: F[A1])(f: A1 => Z, g: Z => A1): F[Z] = xmap(a1, f, g)
+  override def xproduct2[Z, A1, A2](a1: =>F[A1], a2: =>F[A2])(
+    f: (A1, A2) => Z, g: Z => (A1, A2)
+  ): F[Z] = apply2(a1, a2)(f)
+  override def xproduct3[Z, A1, A2, A3](a1: =>F[A1], a2: =>F[A2], a3: =>F[A3])(
+    f: (A1, A2, A3) => Z,
+    g: Z => (A1, A2, A3)
+  ): F[Z] = apply3(a1, a2, a3)(f)
+  override def xproduct4[Z, A1, A2, A3, A4](a1: =>F[A1], a2: =>F[A2], a3: =>F[A3], a4: =>F[A4])(
+    f: (A1, A2, A3, A4) => Z,
+    g: Z => (A1, A2, A3, A4)
+  ): F[Z] = apply4(a1, a2, a3, a4)(f)
+
+  override def xcoproduct1[Z, A1](a1: =>F[A1])(
+    f: A1 => Z,
+    g: Z => A1
+  ): F[Z] = altly1(a1)(f)
+  override def xcoproduct2[Z, A1, A2](a1: =>F[A1], a2: =>F[A2])(
+    f: (A1 \/ A2) => Z,
+    g: Z => (A1 \/ A2)
+  ): F[Z] = altly2(a1, a2)(f)
+  override def xcoproduct3[Z, A1, A2, A3](a1: =>F[A1], a2: =>F[A2], a3: =>F[A3])(
+    f: (A1 \/ (A2 \/ A3)) => Z,
+    g: Z => (A1 \/ (A2 \/ A3))
+  ): F[Z] = altly3(a1, a2, a3)(f)
+  override def xcoproduct4[Z, A1, A2, A3, A4](a1: =>F[A1], a2: =>F[A2], a3: =>F[A3], a4: =>F[A4])(
+    f: (A1 \/ (A2 \/ (A3 \/ A4))) => Z,
+    g: Z => (A1 \/ (A2 \/ (A3 \/ A4)))
+  ): F[Z] = altly4(a1, a2, a3, a4)(f)
+
+  trait AltLaw extends ApplicativeLaw {
+    // TODO: laws
+
+    // <!> is associative:             (a <!> b) <!> c = a <!> (b <!> c)
+    // <$> left-distributes over <!>:  f <$> (a <!> b) = (f <$> a) <!> (f <$> b)
```

Comment:
This is `<|>` now

Reply:
this is direct copypasta from the haskell